db/hints/manager.cc

```diff
@@ -492,9 +492,12 @@ static bool endpoint_downtime_not_bigger_than(const gms::gossiper& gossiper, con
 
     gossiper.for_each_endpoint_state_until(
             [&info] (const gms::inet_address& ip, const gms::endpoint_state& state) {
-        const auto app_state = state.get_application_state_ptr(gms::application_state::HOST_ID);
+        const auto* app_state = state.get_application_state_ptr(gms::application_state::HOST_ID);
+        if (!app_state) {
```

Comment:
let's print an ERROR if we detect a missing HOST ID. It should not happen, and separate issue is needed. Other than that LGTM

Reply:
Good idea, thanks!